### ISSUE TITLE
Provide an API to read/write `Image` pixels with borrows.

### ISSUE THREAD

**Issue by andriyDev** [bevyengine/bevy#15477]
## What problem does this solve or what need does it fill?

Today, the primary (only?) way to access the pixels of an image (reading or writing) is to first remove the `Image` from the `Assets<Image>`, call `Image::try_into_dynamic`, then read or mutate the image, then put the image **back** into the `Assets<Image>` resource.

## What solution would you like?

We can create our own `ImageView` struct which will store a borrow (or mutable borrow) to an `Image` (for "reasonable" image types - most of what `DynamicImage` currently supports). We can then implement the `image::GenericImage` and `image::GenericImageView` traits which will allow the `image` crate to do its job. We can also probably have our own functions `get_color` and `set_color` to make it easier to use with Bevy stuff.

**Comment by Cedar-81** [comment]
Do you mean a trait that makes it possible to read,write and modify images with the image crate?

If that's so i think it shouldn't be hard to implement

(New to contributing please if I am doing something wrong I apologize)

**Comment by andriyDev** [comment]
> Do you mean a trait that makes it possible to read,write and modify images with the image crate?
> 
> If that's so i think it shouldn't be hard to implement
> 
> (New to contributing please if I am doing something wrong I apologize)

The `image` crate already contains the trait we should implement: `image::GenericImageView` and `image::GenericImage`. What we need is something like a wrapper type around `&Image` or `&mut Image` that implements these traits, so we can reuse all the stuff the `image` crate provides, while also avoiding the need to use `ImageBuffer` or `DynamicImage`.

It doesn't seem too hard to implement, I started it a little bit and got a little confused about how the `image` crate expects those traits to be implemented. Feel free to pick this up @Cedar-81!

**Comment by Cedar-81** [comment]
Can I get a link to the repo you implanted this on? I worked with the image crate recently maybe I could help.

**Comment by andriyDev** [comment]
I didn't get very far at all, but here's roughly what I was thinking: https://gist.github.com/andriyDev/c846454c87869963f51131de8ec38bdf

**Comment by andriyDev** [comment]
I think this is basically just fixed by #10392

**Comment by jomala** [comment]
@andriyDev  - What I'd like to be able to do to modify a image using the `image` crate is
```
let buffer: &mut image::ImageBuffer = images.get_mut(&draw_image_id).expect("image not present").into_buffer();
```

That would let me modify the image using methods like `subimage()` and `pixels_mut()` and know they were stored back into `images` without a copy of the whole image being made. I don't think that's possible currently, is it? Using `try_to_dynamic()` copies, and sticking to the `bevy::Image` limits the functionality available.

Am I wrong? If not, can we reopen this issue?
